.gitignore

```diff
@@ -8,3 +8,5 @@
 /.metadata/
 tags
 *~
+
+.vscode/
```

Comment:
You can ignore files/directories locally by adding them to the `.git/info/exclude` file, its contents are merged with the repo `.gitignore` when building the ignore list. There's no reason to add directories to the repo `.gitignore` unless they're likely to be created by something _in_ the repo.

(I mean, it doesn't _hurt_ anything, sure. Just that there's an easy local workaround, only problem is that it's so poorly-communicated it's practically an easter egg. I only learned about it this year, I think.)

Reply:
(There's also a "global" (per-user) `$HOME/.config/git/ignore` file, for anything you want to always ignore in all repos forever.)

Reply:
Good to know about this easter egg. Some git tools include it on the main .gitignore by default